src/system/SystemSockets.h

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file declares the abstraction of socket (file descriptor) events.
+ */
+
+#pragma once
+
+// Include configuration headers
+#include <system/SystemConfig.h>
+
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS
+
+#include <support/BitFlags.h>
+#include <system/SystemError.h>
+
+#include <unistd.h>
+
+namespace chip {
+
+namespace System {
+
+class Layer;
+
+enum class SocketEventFlags : uint8_t
+{
+    kRead   = 0x1, /**< Bit flag indicating if there is a read event on a socket. */
+    kWrite  = 0x2, /**< Bit flag indicating if there is a write event on a socket. */
+    kExcept = 0x4, /**< Bit flag indicating if there is an exceptional condition on a socket (e.g. out-of-band data). */
+    kError  = 0x8, /**< Bit flag indicating if there is an error event on a socket. */
```

Comment:
These are just copied from the current `class SocketEvents`. I personally agree with removing the prefix, but a large majority of enum class currently use it (arguably _also_ my fault, I guess, since I did the conversion from plain enums for many.)